node/grpc/server.go

```diff
@@ -130,6 +132,16 @@ func (s *Server) serveRetrieval() error {
 
 }
 
+func (s *Server) NodeInfo(ctx context.Context, in *pb.NodeInfoRequest) (*pb.NodeInfoReply, error) {
+	memBytes := uint64(0)
+	v, err := mem.VirtualMemory()
+	if err == nil {
+		memBytes = v.Total
+	}
+
+	return &pb.NodeInfoReply{Semver: node.SemVer, Os: runtime.GOOS, Arch: runtime.GOARCH, NumCpu: uint32(runtime.NumCPU()), MemBytes: memBytes}, nil
```

Comment:
I don't think `runtime.NumCPU` is aware of the CPU limits set on the container. 
I think https://github.com/uber-go/automaxprocs can help set the right number of CPUs in `runtime.GOMAXPROCS`.